openstf_dbc/services/predictor.py

```diff
@@ -2,36 +2,137 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
+from enum import Enum
+from typing import List, Optional, Tuple, Union
+
+import pandas as pd
 from openstf_dbc.data_interface import _DataInterface
+from openstf_dbc.services.weather import Weather
+from openstf_dbc.utils.utils import get_datetime_index
+
+
+class PredictorGroups(Enum):
+    MARKET_DATA = "market_data"
+    WEATHER_DATA = "weather_data"
+    LOAD_PROFILES = "LOAD_PROFILES"
 
 
 class Predictor:
-    def get_apx(self, datetime_start, datetime_end):
+    def get_predictors(
+        self,
+        datetime_start,
+        datetime_end,
+        forecast_resolution: Optional[str] = None,
+        location: Union[str, Tuple[float, float]] = None,
+        predictor_groups: Optional[List[str]] = None,
```

Comment:
Would be nice to include (here or in the docstring) which options are available
Also, do we want a list of str as input, or a list of PredictorGroups.keys?

Reply:
I already noticed this one. I've updated the typehinting and implementation to accept both: `Union[List[PredictorGroups], List[str], None]`. Perhaps better to refer to the enum for the allowed values to have a single source of truth. update: I've made the docstring more clear.